### PR TITLE
[12.0][ADD] helpdesk_mgmt_ticket_start_date

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Add an editable Start Date field that initializes with today's date

### CODE REVIEW

File: helpdesk_mgmt_ticket_start_date/README.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+========================
+Helpdesk Ticket Reminder
+========================
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_start_date/README.rst]: Please update the whole `README.rst` file

File: helpdesk_mgmt_ticket_start_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Helpdesk Ticket Start Date",
+    "summary": """
+        Add an editable Start Date field that initializes with today's date
+    """,
+    "version": "12.0.2.1.0",
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_start_date/__manifest__.py]: Wrong version:
```suggestion
    "version": "12.0.1.0.0",
```

File: helpdesk_mgmt_ticket_start_date/views/helpdesk_ticket_views.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    </record>
+
+    <record id="ticket_view_tree" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.tree (start_date)</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='create_date']" position="after">
+                <field name="start_date"/>
+            </xpath>
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_start_date/views/helpdesk_ticket_views.xml:24]: Could you add making invisible `create_date`?

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review. When comments are attended, I'll test it

**Comment by dalonsod** [comment]
Locally tested, it works :)

Could you add an init hook in order to initialize `start_date` field of existing tickets with their `create_date`?

**Comment by ChristianSantamaria** [comment]
The init_hook is not running and I don't know why

@dalonsod Could you take a look at it?

File: helpdesk_mgmt_ticket_start_date/hooks.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+from odoo import api, SUPERUSER_ID
+
+
+def post_init_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    HelpdeskTicket = env['helpdesk.ticket']
+    rec = HelpdeskTicket.search([('start_date', '=', False)])
+
+    for req in rec:
+        rec.start_date = rec.create_date
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_start_date/hooks.py]: When you install, an error is thrown (an "expected singleton" error).
This worked for me (variables were wrong assigned):
```suggestion
    tickets = HelpdeskTicket.search([('start_date', '=', False)])

    for rec in tickets:
        rec.start_date = rec.create_date
```

**Review by dalonsod — CHANGES_REQUESTED** [review]

**Review by dalonsod — APPROVED** [review]
LGTM
@lmiguens-solvos could you review? Thanks!

**Comment by lmiguens-solvos** [comment]
It works well for me, although I can think of 2 improvements (or questions). 
1.- To be able to modify this date when registering the ticket. Why is it not allowed to modify it? Is it complicated?
and 
2.- Hide the creation date in form view. Perhaps it is no longer necessary having this new start date. It would already be seen only as metadata, as in other entities.

**Comment by dalonsod** [comment]
> 1.- To be able to modify this date when registering the ticket. Why is it not allowed to modify it? Is it complicated?

It's possible, the current implementation was defined by me.

> 2.- Hide the creation date in form view. Perhaps it is no longer necessary having this new start date. It would already be seen only as metadata, as in other entities.

I agree.

**Comment by lmiguens-solvos** [comment]
I have already seen that my 2 requests are resolved. Perfect! Thank you!
But now I have a another question @dalonsod 
My new question is about the ticket numbering: if the year was included in the ticket number, should it be created with the year of this new registration date? or will it be too complicated

**Comment by dalonsod** [comment]
> My new question is about the ticket numbering: if the year was included in the ticket number, should it be created with the year of this new registration date? or will it be too complicated

It seems to be possible, but complicated.

Looking at https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_sequence.py#L103 , it's possible to obtain the proper sequence range depending on a custom date. But I should investigate in depth how to properly call it in this helpdesk context

File: helpdesk_mgmt_ticket_start_date/models/helpdesk_ticket.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import fields, models, api
+
+import datetime
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    start_date = fields.Datetime()
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_start_date/models/helpdesk_ticket.py]: @ChristianSantamaria  One last suggested change, in order to log modifications:
```suggestion
    start_date = fields.Datetime(track_visibility="onchange")
```

**Comment by lmiguens-solvos** [comment]
> It seems to be possible, but complicated.

ok, then surely it is not necessary to do so

**Comment by dalonsod** [comment]
Good!
Please squash commits and, if @lmiguens-solvos agrees, I'll merge it

**Review by lmiguens-solvos — APPROVED** [review]
LGTM

**Comment by dalonsod** [comment]
Merging...

**Pull request merged**